de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/project/FileSystemChangeListener.java

```diff
@@ -398,52 +398,55 @@ public void propertyChanged(
         }
     }
 
+    /**
+     * {@inheritDoc}
+     * <p></p>
+     * Generates and dispatches creation activities for copied files. Copied
+     * directories are handled by {@link #fileCreated(VirtualFileEvent)} and
+     * contained files are subsequently handled by this listener.
+     *
+     * @param virtualFileCopyEvent {@inheritDoc}
+     */
     @Override
     public void fileCopied(
-        @NotNull
-        VirtualFileCopyEvent virtualFileCopyEvent) {
-        if (!enabled) {
-            return;
-        }
+            @NotNull
+                    VirtualFileCopyEvent virtualFileCopyEvent) {
 
-        VirtualFile virtualFile = virtualFileCopyEvent.getFile();
-        File newFile = new File(virtualFile.getPath());
+        assert enabled : "the file copied listener was triggered while it was disabled";
 
-        if (incomingFilesToFilterFor.remove(newFile)) {
-            return;
-        }
+        VirtualFile copy = virtualFileCopyEvent.getFile();
 
-        IPath path = IntelliJPathImpl.fromString(newFile.getPath());
-        IntelliJProjectImpl project = getProjectForResource(path);
+        if (copy.isDirectory()) {
```

Comment:
Why not as assert?

Reply:
I was not sure whether this should be an assert or not. This check is supposed to warn us if IntelliJ changes its handling of the `fileCopied(...)` listener. This would only really make a difference if we decide to disable assertions when distributing the plugin. But, since a lot of the nullchecks are done through assertions as well, it may be a good idea to leave them enabled anyway to make debugging a nullpointer exception easier (since the assertions show us a more precise location for where the violation actually happened).

Will change it to an assertion.